src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceValueSerializer.cs

```diff
@@ -137,26 +137,30 @@ private ODataResourceValue CreateODataResourceValue(object graph, IEdmStructured
             }
             else
             {
-                HashSet<string> structuralPropertyNames = new HashSet<string>();
-
-                foreach(IEdmStructuralProperty structuralProperty in expectedType.DeclaredStructuralProperties())
+                // If a type is an org.OData.Core.V1.DataModificationExceptionType type, 
+                // then we include all it's properties and not just the structural properties.
+                if (expectedType.Definition.FullTypeName().Equals("Org.OData.Core.V1.DataModificationExceptionType"))
```

Comment:
This doesn't look like a clean way to check the exception type. Is there a better way?

Reply:
We are checking the name of the `IEdmStructuredTypeReference`.